lib/chef/provider/package/chocolatey.rb

```diff
@@ -136,14 +136,27 @@ def check_resource_semantics!
         #
         # @return [String] full path of choco.exe
         def choco_exe
-          @choco_exe ||=
+          @choco_exe ||= begin
+            path = choco_install_path
+
+            # GetEnvironmentVariable returns "" on failure.
+            if path.to_s.empty?
+              raise Chef::Exceptions::MissingLibrary.new("Could not find Chocolatey. Is it installed?")
+            end
+
             ::File.join(
-              powershell_out!(
-                "[System.Environment]::GetEnvironmentVariable('ChocolateyInstall', 'MACHINE')"
-              ).stdout.chomp,
+              path,
               "bin",
               "choco.exe",
-          )
+              )
+          end
+        end
+
+        # lets us mock out an incorrect value for testing.
+        def choco_install_path
+          powershell_out!(
```

Comment:
should `@choco_install_path ||=` this so you don't have to create local variables to protect against double-invocation elsewhere...